### PR TITLE
Add utility to migrate from ExamLog and ExamAttemptLog to consolidated logging.

### LINKED ISSUE #8477: Write utility function to convert quiz progress logged with ExamLog and ExamAttemptLog to MasteryLog and AttemptLog
```
    Proposed Mapping:

    ExamLog ---------> MasteryLog + ContentSessionLog + ContentSummaryLog

    Fields:
    exam_id ---------> content_id (ContentSessionLog + ContentSummaryLog)
    user    ---------> user (ContentSessionLog + ContentSummaryLog)
    closed  ---------> complete (MasteryLog), progress=1 (ContentSessionLog + ContentSummaryLog)
    completion_timestamp ---------> completion_timestamp (MasteryLog + ContentSummaryLog),
                                    end_timestamp (MasteryLog + ContentSessionLog + ContentSummaryLog)

    ExamAttemptLog --> AttemptLog

    Fields:
    examlog ---------> masterylog
    content_id ------> combine with <item> as <content_id>:<item> to give a composite item id

    All other ExamAttemptLog fields are shared with AttemptLog

    For MasteryLog + ContentSessionLog + ContentSummaryLog
    infer start_timestamp from Min across ExamAttemptLogs
    
    mastery_criterion (MasteryLog), { "type": "quiz" }
    mastery_level (MasteryLog), short integer derived deterministically from exam_id
    channel_id (ContentSessionLog + ContentSummaryLog), None
    kind (ContentSessionLog + ContentSummaryLog), quiz
```

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* Implements a utility to map ExamLog and ExamAttemptLog to our more general Content logging models
* Removes restrictive validation on MasteryLog mastery_level to allow mastery levels that will prevent syncing to historic versions of Kolibri
* Allows channel_id to be nullable on ContentSession and ContentSummary log models
* Adds tests for the function

## References
Fixes #8477

## Reviewer guidance
Can you think of any edge cases not covered by the tests?

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [x] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/logger/utils/exam_log_migration.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+            .first()
+        )
+    for log in logs:
+        if log.item in existing_log_items:
+            existing_log = existing_log_items[log.item]
+            # Last write wins
+            # Otherwise we ignore the updated log.
+            if existing_log.end_timestamp < AttemptLog._meta.get_field(
+                "end_timestamp"
+            ).from_db_value(log.end_timestamp, None, None, None):
```
Comment (bjester) [inline kolibri/core/logger/utils/exam_log_migration.py:96]: Why is this `AttemptLog._meta.get_field( [...]` stuff required?

File: kolibri/core/logger/utils/exam_log_migration.py [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    mastery_criterion = {"type": content_kinds.QUIZ}
+
+    attempts_blocklist = {
+        "id",
+        "_morango_dirty_bit",
+        "_morango_source_id",
+        "_morango_partition",
+        "examlog_id",
+        "content_id",
+    }
```
Comment (bjester) [inline kolibri/core/logger/utils/exam_log_migration.py:176]: Is this related to `attempt_log_fields`? Are they complimentary?

File: kolibri/core/logger/utils/exam_log_migration.py (hunk 1 above)
Comment (rtibbles) [reply kolibri/core/logger/utils/exam_log_migration.py:96]: Oh sorry, I'll add an explanatory comment - without this, it's not a datetime, but the serialized string form (as Morango deserialize doesn't run through field `to_python_value`) so I needed to do this to coerce it to a datetime for the comparison.

File: kolibri/core/logger/utils/exam_log_migration.py (hunk 2 above)
Comment (rtibbles) [reply kolibri/core/logger/utils/exam_log_migration.py:176]: Yes, I added this first - but I think I can probably remove this in favour of just iterating over `attemptlog_fields` below instead.

Comment (rtibbles) [reply kolibri/core/logger/utils/exam_log_migration.py:176]: When I came back to this, I realized that I can't unify them - I've clarified the function with naming and added some comments.

**Review by bjester — APPROVED** [review]

**Comment by rtibbles** [comment]
Going to merge as the only failure is the Python linting failure that is now fixed in mainline.
